sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1931,6 +1931,42 @@ object functions extends LegacyFunctions {
     new Murmur3Hash(cols.map(_.expr))
   }
 
+  /**
+   * Encrypts input using AES and Returns the result as a binary column.
+   * Key lengths of 128, 192 or 256 bits can be used. 192 and 256 bits keys can be used if Java
+   * Cryptography Extension (JCE) Unlimited Strength Jurisdiction Policy Files are installed. If
+   * either argument is NULL, the result will also be null. If input is invalid, key length is not
+   * one of the permitted values or using 192/256 bits key before installing JCE, an exception will
+   * be thrown.
+   *
+   * @param input binary column to encrypt input
+   * @param key binary column of 128, 192 or 256 bits key
+   *
+   * @group misc_funcs
+   * @since 2.0.0
+   */
+  def aes_encrypt(input: Column, key: Column): Column = withExpr {
```

Comment:
Actually we do have a mess of it. Generally we have 3 kinds of parameters: `Column`, column name string, literal. For example, `pow` takes 2 parameters, and we have 8 overloaded versions of it, which are all combinations of these 3 kinds of parameters except both literals. However, some functions like `pmod`, only have both `Column` version. Some functions like `sha2` only have `Column`, literal combination.

Personly I think for something like `sha2` and this one, where one parameter will be literal 99% of the time, we should just give a `Column`, literal combination. i.e. `def aes_encrypt(input: Column, key: String)`. For something like `pow`, we only need to provide the both `Column` version, as column name string and literal are easy to parse to `Column`, by `col()` and `lit()`.